client/order/bookside_test.go

```diff
@@ -0,0 +1,1311 @@
+						},
+					},
+					2: {
+						&Order{
+							OrderID:  [32]byte{'c'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 2,
+							Rate:     2,
+							Time:     10,
+						},
+					},
+				},
+				makeRateIndex([]uint64{1, 2}),
+				ascending,
+			),
+			entry: &Order{
+				OrderID:  [32]byte{'c'},
+				Side:     msgjson.BuyOrderNum,
+				Quantity: 2,
+				Rate:     2,
+				Time:     10,
+			},
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						&Order{
+							OrderID:  [32]byte{'a'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 10,
+							Rate:     1,
+							Time:     10,
+						},
+						&Order{
+							OrderID:  [32]byte{'b'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 5,
+							Rate:     1,
+							Time:     10,
+						},
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			wantErr: false,
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						&Order{
+							OrderID:  [32]byte{'a'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 10,
+							Rate:     1,
+							Time:     10,
+						},
+						&Order{
+							OrderID:  [32]byte{'b'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 5,
+							Rate:     1,
+							Time:     10,
+						},
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: &Order{
+				OrderID:  [32]byte{'c'},
+				Side:     msgjson.BuyOrderNum,
+				Quantity: 2,
+				Rate:     3,
+				Time:     10,
+			},
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						&Order{
+							OrderID:  [32]byte{'a'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 10,
+							Rate:     1,
+							Time:     10,
+						},
+						&Order{
+							OrderID:  [32]byte{'b'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 5,
+							Rate:     1,
+							Time:     10,
+						},
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			wantErr: true,
+		},
+		{
+			side: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						&Order{
+							OrderID:  [32]byte{'a'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 10,
+							Rate:     1,
+							Time:     10,
+						},
+						&Order{
+							OrderID:  [32]byte{'b'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 5,
+							Rate:     1,
+							Time:     10,
+						},
+						&Order{
+							OrderID:  [32]byte{'c'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 2,
+							Rate:     1,
+							Time:     10,
+						},
+					},
+				},
+				makeRateIndex([]uint64{1}),
+				ascending,
+			),
+			entry: &Order{
+				OrderID:  [32]byte{'b'},
+				Side:     msgjson.BuyOrderNum,
+				Quantity: 5,
+				Rate:     1,
+				Time:     10,
+			},
+			expected: makeBookSide(
+				map[uint64][]*Order{
+					1: {
+						&Order{
+							OrderID:  [32]byte{'a'},
+							Side:     msgjson.BuyOrderNum,
+							Quantity: 10,
+							Rate:     1,
+							Time:     10,
+						},
```

Comment:
You have this exact same order defined separately 16 times. Many of the others have similar numbers of identical copies. Each of those is 7 lines. It's okay to be verbose, but it can become difficult to read. I would suggest you either 1) create a single set of orders to reuse for all of the tests, or 2) write a little function to generate the orders. 

`func newTestOrder(idStr string, sell bool, qty, rate, stamp uint64) *Order`

Reply:
sorry about that, thanks for pointing it out. I couldn't see it somehow.